src/Libraries/Microsoft.Extensions.Diagnostics.Probes/TcpEndpointOptions.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Shared.Data.Validation;
+
+namespace Microsoft.Extensions.Diagnostics.Probes;
+
+/// <summary>
+/// Options to control TCP-based health check probes.
+/// </summary>
+[SuppressMessage("Major Code Smell", "S109:Magic numbers should not be used", Justification = "In place numbers make the ranges cleaner")]
+[SuppressMessage("Design", "CA1034:Nested types should not be visible", Justification = "It's fine")]
+public class TcpEndpointOptions
```

Comment:
@geeknoid perhaps a question to answer during your API reviews meeting: Should this class be named `TcpEndpointHealthCheckOptions` to match other file names?

![classes](https://github.com/dotnet/extensions/assets/147369450/1b111b71-369d-41e8-ace8-6bc8446bc897)

Or as this class is in the namespace `Microsoft.Extensions.Diagnostics.Probes`, it is enough to indicate that these are the options for the health check probe perhaps.